compiler/cpp/src/thrift/generate/t_js_generator.cc

```diff
@@ -1134,7 +1156,12 @@ void t_js_generator::generate_process_function(t_service* tservice, t_function*
   f_service_ << indent() << "if (this._handler." << tfunction->get_name()
              << ".length === " << fields.size() << ") {" << endl;
   indent_up();
-  indent(f_service_) << "Q.fcall(this._handler." << tfunction->get_name() << ".bind(this._handler)";
```

Comment:
This is also fixing a bug in the es6 compiler which references the `Q` package without importing it. ES6 should use the native `Promise.resolve` instead.